Makefile.envs

```diff
@@ -148,6 +149,7 @@ export MAIN_MODULE_LDFLAGS= $(LDFLAGS_BASE) \
 	-lwebsocket.js \
 	-leventloop.js \
 	-lhiwire \
+	-lHacl_Hash_SHA2 \
```

Comment:
Maybe we can update [`Setup.local`](https://github.com/pyodide/pyodide/blob/main/cpython/Setup.local) such as:

```
_sha2 sha2module.c -I_hacl/include Modules/_hacl/libHacl_Hash_SHA2.a
```

instead of linking it here manually?

Reply:
I'll fix this in a followup PR with a bit more `cpython/Makefile` tidying.